compiler/rustc_trait_selection/src/error_reporting/traits/suggestions.rs

```diff
@@ -4668,9 +4668,8 @@ impl<'a, 'tcx> TypeErrCtxt<'a, 'tcx> {
                 && b.expr.is_none()
             {
                 sugg_spans.push((
-                    // The span will point to the closing curly brace `}` of the block.
-                    b.span.shrink_to_hi().with_lo(b.span.hi() - BytePos(1)),
-                    "\n    Ok(())\n}".to_string(),
+                    self.tcx.sess.source_map().end_point(b.span).shrink_to_lo(),
+                    "\n    Ok(())\n".to_string(),
```

Comment:
Could you fix the indentation level too? 
And maybe avoid creating an empty line before the Ok?

Reply:
> Could you fix the indentation level too? And maybe avoid creating an empty line before the Ok?

Done. Thank you.